lib/pharos/phase.rb

```diff
@@ -80,8 +80,23 @@ def host_configurer
 
     # @return [Pharos::SSH::Client]
     def master_ssh
-      return cluster_context['master-ssh'] if cluster_context['master-ssh']
-      fail "Phase #{self.class.name} does not have master ssh"
+      master_hosts = @config.master_hosts
+      master_host = master_hosts.first
+
+      # Select non-self master host when available
+      if master_hosts.size > 1 && master_host == host
+        alt_masters = master_hosts[1..-1].select(&:master_valid?)
+        master_host = alt_masters.first unless alt_masters.empty?
```

Comment:
What happens when we don't find "valid" master here?

Reply:
The original (self) master will be used (assigned above)

Reply:
This might be completely useless. The assumption is that the master does work before and after reboot and is capable of `kubectl drain`ing and `uncordon`ing itself, as I'm testing this on one master anyway.

Reply:
Yes I get that... but does it work if "self" is used?

Reply:
It does, I removed the whole trickery here.